features/step_definitions/user_steps.rb

```diff
@@ -151,26 +158,17 @@ def page_heading_text(page)
   click_on('Register')
 end
 
-Given('they login as {string}') do |username|
-  fill_in('username', with: username)
-  fill_in('password', with: 'bar')
-  click_on('SignIn')
-  click_on('I Agree')
-end
+Given(/^they login as "(.*)"( with a random pid)?$/) do |user_string, with_random_pid|
+  if user_string == "the newly registered user"
+    user_string = @username
+  end
 
-Given('they login as the newly registered user') do
-  fill_in('username', with: @username)
-  fill_in('password', with: 'bar')
-  click_on('SignIn')
-  click_on('I Agree')
-end
+  log_in_as(user_string)
+
+  if with_random_pid == ' with a random pid'
```

Comment:
You don't need the surrounding `if` statement since you have the `if with_random_pid` check appended at the end of the line, which is enough.

Reply:
@CharlesIC Thanks. Change has been made accordingly. Please re-review.